### ISSUE TITLE
Food Demand Forecasting for Restaurants

### ISSUE THREAD

**Issue by AyushB21** [florinpop17/app-ideas#878]
# Food Demand Forecasting for Restaurants

**Tier:** 2-Advanced

Food demand forecasting for restaurants is an application built to estimate the food demand that a restaurant is expected to receive within a day, or a week or a month. The historical information concerning the previous sales and the outside influences such as observed festivities, weather and promotions that an application aids the restaurant managers to organize the supplies, control on wastage and personnel optimum.

Answer the following questions:

- **What is the purpose of this application?**

  The primary goal of this application is to help restaurant owners and managers to forecast food requirements in a better manner. It assists to avoid preparing extra portions of ingredients, using too little of it, slashing potential loss, and making work smoothly.

- **Are there any resources needed in order to complete the project?**

  1. Using historical trends about sales / from POS system of the restaurant.
  2. Weather API’s and holiday calendars are some of the examples of the external data source.
  3. Interpreting the results from data using machine learning libraries (TensorFlow, scikit-learn) in demand forecasting.

## User Stories

- [ ] User has the ability to feed historical sales numbers for the purpose of training the forecasting model.
- [ ] User is able to see a panel indicating forecast of the consumed food for a particular day or a week.
- [ ] User of the application can see the charts and graphs in the form of bar graphs and pie charts depicting the sales history and predicted requirements.
- [ ] User can superimpose other factors such as holidays or special occasions to be factored into demand forecast.
- [ ] User also has possibility to get alerts or notifications regarding possible stock shortage according to the predictions.
- [ ] It can help user download the report of expected demand to meet the purpose of using in managing inventory.

## Bonus features

- [ ] User can push real time weather data to update the forecasts on the go.
- [ ] They also allow the user to compare actual sales, or turnover as it is preferred in certain geographies, with forecasted demand to better predict future sales.
- [ ] With integrated models, the user can set a “What-if” environment to make users evaluate the effects of a shift in price or the introduction of a promotion.
- [ ] User can have an AI agent for recommendation on buying inventory.

## Useful links and resources

- [Machine Learning for Time Series Forecasting](https://www.tensorflow.org/tutorials/structured_data/time_series)
- [Restaurant Demand Forecasting Using Machine Learning](https://www.sciencedirect.com/science/article/pii/S2212827119301568)
- [scikit-learn Documentation](https://scikit-learn.org/stable/)
- [Weather API for External Data](https://openweathermap.org/api)

## Example projects

- [Uber Eats Restaurant Demand Prediction](https://merchants.ubereats.com/se/en/resources/articles/restaurant-data-analytics/)
- [AI for Food Demand Forecasting](https://www.5out.io/post/the-new-trends-of-dining-ai-demand-forecasting-for-restaurants)
- [Time Series Forecasting with Prophet](https://www.kaggle.com/code/prashant111/tutorial-time-series-forecasting-with-prophet)

The following application will enable a restaurant manager to have the tools of evaluation to forecast and control the food demand effectively.

**Comment by AyushB21** [comment]
New app idea for restaurant food demand forecasting prediction for better management.

**Comment by anu098jaiswal** [comment]
do u have any idea that this repo is added in hacktoberfets or nt? as most of repos shows repo excluded after creating pr

**Comment by AyushB21** [comment]
> do u have any idea that this repo is added in hacktoberfets or nt? as most of repos shows repo excluded after creating pr

yes it is added i feel those not following templates are mainly closed.
